control-plane/connect-inject/endpoints_controller.go

```diff
@@ -32,6 +32,7 @@ const (
 	MetaKeyKubeServiceName     = "k8s-service-name"
 	MetaKeyKubeNS              = "k8s-namespace"
 	MetaKeyManagedBy           = "managed-by"
+	MetaKeySyntheticNode       = "synthetic-node"
```

Comment:
It should be prefixed with `consul-`
```suggestion
	MetaKeySyntheticNode       = "consul-synthetic-node"
```

Reply:
Ashwin and I synced up, and he said that we can't use because it's reserved. Resolving this!